src/Integration.Vsix/Analysis/IssueConsumerFactory_IssueHandler.cs

```diff
@@ -76,9 +82,20 @@ internal class IssueHandler
                 // The text buffer might have changed since the analysis was triggered, so translate
                 // all issues to the current snapshot.
                 // See bug #1487: https://github.com/SonarSource/sonarlint-visualstudio/issues/1487
-                var translatedIssues = translateSpans(issues, textDocument.TextBuffer.CurrentSnapshot);
-
-                var newSnapshot = new IssuesSnapshot(projectName, projectGuid, textDocument.FilePath, translatedIssues);
+                var translatedIssues = translateSpans(issues, textDocument.TextBuffer.CurrentSnapshot).ToList();
+                
+                localHotspotsStore.AddOrUpdate(textDocument.FilePath,
+                    translatedIssues
+                        .Where(issue => (issue.Issue as IAnalysisIssue)?.Type == AnalysisIssueType.SecurityHotspot));
+                
+                var newSnapshot = new IssuesSnapshot(projectName,
+                    projectGuid,
+                    textDocument.FilePath,
+                    translatedIssues.Where(issue =>
+                    {
+                        var analysisIssueType = (issue.Issue as IAnalysisIssue)?.Type;
+                        return analysisIssueType == AnalysisIssueType.Bug || analysisIssueType == AnalysisIssueType.CodeSmell;
```

Comment:
This would be clearer and shorter checking for `!= AnalysisIssueType.SecurityHotspot`.
That would make it clearer that there aren't any other issues types that we are deliberately filtering out.